[Task]: Add additional native-image configs related to XML

### Description

Native-image fails at runtime for the following BBE : [`XML Operations`](https://ballerina.io/learn/by-example/xml-operations/) with lang `2201.3.0-rc1`

Sample :

```ballerina
public function main() returns error? {
    xml x1 = xml `<name>Sherlock Holmes</name>`;
    xml:Element x2 = 
        xml `<details>
                <author>Sir Arthur Conan Doyle</author>
                <language>English</language>
            </details>`;


    xml x3 = x1 + x2;

    xml x4 = xml `<name>Sherlock Holmes</name><details>
                        <author>Sir Arthur Conan Doyle</author>
                        <language>English</language>
                  </details>`;

    boolean _ = x3 == x4;

    xml x5 = xml `<para id="greeting">Hello</para>`;
    string _ = check x5.id;

    string? _ = check x5?.name;

    x2.setChildren(xml `<language>French</language>`);
}
```

Runtime error :
```console
[2022-10-24 19:25:21,855] SEVERE {b7a.log.crash} - null 
java.lang.ExceptionInInitializerError
        at java.lang.Class.ensureInitialized(DynamicHub.java:525)
        at org.apache.axiom.om.OMAbstractFactory.<clinit>(OMAbstractFactory.java:106)
        at io.ballerina.runtime.internal.values.XmlNonElementItem.<init>(XmlNonElementItem.java:47)
        at io.ballerina.runtime.internal.values.XmlText.<init>(XmlText.java:39)
        at io.ballerina.runtime.internal.XmlFactory.createXMLText(XmlFactory.java:383)
        at io.ballerina.runtime.internal.XmlFactory.createXMLText(XmlFactory.java:393)
        at user.xml_test/0.main.main(main.bal:4)
        at user.xml_test/0.$_init.$lambda$main$(xml_test)
        at io.ballerina.runtime.internal.scheduling.SchedulerItem.execute(SchedulerItem.java:54)
        at io.ballerina.runtime.internal.scheduling.Scheduler.run(Scheduler.java:324)
        at io.ballerina.runtime.internal.scheduling.Scheduler.runSafely(Scheduler.java:291)
        at java.lang.Thread.run(Thread.java:829)
        at com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:705)
        at com.oracle.svm.core.posix.thread.PosixPlatformThreads.pthreadStartRoutine(PosixPlatformThreads.java:202)
Caused by: org.apache.commons.logging.LogConfigurationException: java.lang.ClassNotFoundException: org.apache.commons.logging.impl.LogFactoryImpl (Caused by java.lang.ClassNotFoundException: org.apache.commons.logging.impl.LogFactoryImpl)
        at org.apache.commons.logging.LogFactory.createFactory(LogFactory.java:1276)
        at org.apache.commons.logging.LogFactory$2.run(LogFactory.java:1065)
        at java.security.AccessController.executePrivileged(AccessController.java:169)
        at java.security.AccessController.doPrivileged(AccessController.java:83)
        at org.apache.commons.logging.LogFactory.newFactory(LogFactory.java:1062)
        at org.apache.commons.logging.LogFactory.getFactory(LogFactory.java:650)
        at org.apache.commons.logging.LogFactory.getLog(LogFactory.java:685)
        at org.apache.axiom.locator.PriorityBasedOMMetaFactoryLocator.<clinit>(PriorityBasedOMMetaFactoryLocator.java:31)
        ... 14 more
Caused by: java.lang.ClassNotFoundException: org.apache.commons.logging.impl.LogFactoryImpl
        at jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:52)
        at jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:133)
        at org.apache.commons.logging.LogFactory.createFactory(LogFactory.java:1131)
        ... 21 more
```

This is previously checked with lang and was working with some additional configurations. But since the **axiom version is upgraded** recently in ballerina lang-runtime, this BBE is failing again due to some missing configurations.

After running with the tracing-agent, the build succeed. So we need to add these additional configs to ballerina-runtime `META-INF` directory.

### Describe your task(s)

Obtain the required configurations from the tracing agent and add them to lang-runtime

### Related area

-> Runtime

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

--- a/bvm/ballerina-runtime/src/main/resources/META-INF/native-image/org.ballerinalang/ballerina-runtime/native-image.properties
+++ b/bvm/ballerina-runtime/src/main/resources/META-INF/native-image/org.ballerinalang/ballerina-runtime/native-image.properties
@@ -1,0 +1,17 @@
+# Copyright (c) 2022, WSO2 LLC. (https://www.wso2.com) All Rights Reserved.
+#
+# WSO2 LLC. licenses this file to you under the Apache License,
+# Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+Args = -H:MaxDuplicationFactor=40.0

--- a/bvm/ballerina-runtime/src/main/resources/META-INF/native-image/org.ballerinalang/ballerina-runtime/reflect-config.json
+++ b/bvm/ballerina-runtime/src/main/resources/META-INF/native-image/org.ballerinalang/ballerina-runtime/reflect-config.json
@@ -1,0 +1,63 @@
+[
+    {
+        "name": "org.apache.axiom.locator.DefaultOMMetaFactoryLocator",
+        "methods": [
+            {
+                "name": "<init>",
+                "parameterTypes": []
+            }
+        ]
+    },
+    {
+        "name": "org.apache.axiom.om.impl.llom.factory.AxiomNodeFactoryImpl",
+        "fields": [
+            {
+                "name": "INSTANCE"
+            }
+        ]
+    },
+    {
+        "name": "org.apache.axiom.om.impl.llom.factory.OMLinkedListMetaFactoryLoader",
+        "methods": [
+            {
+                "name": "<init>",
+                "parameterTypes": []
+            }
+        ]
+    },
+    {
+        "name": "org.apache.commons.logging.LogFactory"
+    },
+    {
+        "name": "org.apache.commons.logging.impl.Jdk14Logger",
+        "methods": [
+            {
+                "name": "<init>",
+                "parameterTypes": [
+                    "java.lang.String"
+                ]
+            }
+        ]
+    },
+    {
+        "name": "org.apache.commons.logging.impl.Log4JLogger"
+    },
+    {
+        "name": "org.apache.commons.logging.impl.LogFactoryImpl",
+        "methods": [
+            {
+                "name": "<init>",
+                "parameterTypes": []
+            }
+        ]
+    },
+    {
+        "name": "org.apache.commons.logging.impl.WeakHashtable",
+        "methods": [
+            {
+                "name": "<init>",
+                "parameterTypes": []
+            }
+        ]
+    }
+]

--- a/bvm/ballerina-runtime/src/main/resources/META-INF/native-image/org.ballerinalang/ballerina-runtime/resource-config.json
+++ b/bvm/ballerina-runtime/src/main/resources/META-INF/native-image/org.ballerinalang/ballerina-runtime/resource-config.json
@@ -1,0 +1,18 @@
+{
+    "resources": {
+        "includes": [
+            {
+                "pattern": "\\QMETA-INF/axiom.xml\\E"
+            }
+        ]
+    },
+    "bundles": [
+        {
+            "name": "MessagesBundle",
+            "locales": [
+                "",
+                "en-US"
+            ]
+        }
+    ]
+}

